File: android/LiveTales/app/src/main/java/com/zed/livetales/managers/LanguageMgr.java
package com.zed.livetales.managers;

/**
 * Created by jemalpartida on 25/11/2016.
 */

public class LanguageMgr
{
    public static LanguageMgr getInstance()
    {
        if( LanguageMgr.instance == null )
        {
            LanguageMgr.instance = new LanguageMgr();
        }

        return LanguageMgr.instance;
    }

    public String getId()
    {
        return this.langId;
    }

    public void setId(String langId)
    {
        boolean supported = false;
        for( String id :SUPPORTED_LANGS )
        {
            if( id.equals( langId ) )
                supported = true;
        }

        this.langId = ( supported ) ? langId : DEFAULT_LANG;
    }


    /*============================================================================================*/
    /*                                    Private Section                                         */
    /*============================================================================================*/
    private LanguageMgr()
    {
        this.langId = DEFAULT_LANG;
    }


    private static LanguageMgr instance;

    private String langId;

    private final String DEFAULT_LANG = "en";
    private final String[] SUPPORTED_LANGS = {"en","es","de"};
}

File: ios/LiveTales-smartphones/uoat/LanguageMgr.swift
//
//  LanguageMgr.swift
//  uoat
//
//  Created by Pyro User on 5/5/16.
//  Copyright © 2016 Zed. All rights reserved.
//

import Foundation

class LanguageMgr
{
    static let getInstance = LanguageMgr();
    
    func getId() -> String
    {
        return self.langId;
    }
    
    func setId( langId:String )
    {
        self.langId = ( SUPPORTED_LANGS.contains( langId ) ) ? langId : DEFAULT_LANG;
    }
    
    
    /*=============================================================*/
    /*                       Private Section                       */
    /*=============================================================*/
    private  init()
    {
        self.langId = DEFAULT_LANG;
    }
    
    private var langId:String;
    private let SUPPORTED_LANGS = ["en","es","de"];
    private let DEFAULT_LANG:String = "en";
}